plugin_tooling/src/melnorme/lang/tooling/structure/GlobalSourceStructure.java

```diff
@@ -0,0 +1,49 @@
+package melnorme.lang.tooling.structure;
+
+import static java.util.Comparator.comparing;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import melnorme.utilbox.collections.ArrayList2;
+import melnorme.utilbox.misc.Location;
+
+public class GlobalSourceStructure {
+	private static final SortedMap<Location, SortedSet<StructureElement>> aggregatedElements = new TreeMap<>(comparing(Location::toPath));
+	
+	private static final EnumSet<StructureElementKind> HIDDEN_ELEMENT_KINDS = EnumSet.of(StructureElementKind.EXTERN_CRATE,

```

Comment:
Not sure if we should hide methods, impls, etc. as well. It's clearly something we should discuss about.